console-subscriber/src/aggregator.rs

```diff
@@ -295,6 +296,12 @@ impl Aggregator {
                             stats.wakes += 1;
                             stats.last_wake = Some(at);
 
+                            // If currently "inside" this task's poll, then the
+                            // task has woken itself.
+                            if stats.current_polls > 0 {
+                                stats.self_wakes += 1;
+                            }
```

Comment:
hmm, is this accurate? 

the task might be being polled on one worker thread, and woken from a task on _another_ worker. if we determine a wake to be a "self-wake" any time a task is woken while it is being polled, we will count wakes from another task that occur concurrently with a poll.

i'm not sure how frequently this will cause false positives, so it may not be a huge deal, but i felt like it was worth pointing out.

Reply:
Oh yes, true. I suppose the proper way would be to crawl the event `parent`s. Or, or do something with a thread local in the subscriber...